app/builtin-pages/com/settings/favicon-picker.js

```diff
@@ -80,4 +81,9 @@ async function onClickRemove () {
   selectedFavicon = null
   onSelect(null)
   rerender()
+}
+
+async function uploadFavicon () {
+  onSelect(await beaker.browser.uploadFavicon())
```

Comment:
Right now, if you cancel, `onSelect()` gets called with a falsy value and any existing favicon is removed. Suggest changing to this to solve that:

```suggestion
var v = await beaker.browser.uploadFavicon()
if (v) onSelect(v)
```